specs/latest/2.0/index.html

```diff
@@ -1193,22 +1193,33 @@ <h2 class="no-toc">Table of contents</h2>
         </p>
       </dt>
       <dd>
-        If <code>length</code> is 0, it defaults to <code>dstBuffer.length - dstOffset</code>.
-        For the WebGLBuffer bound to the passed target, read <code>length</code>
-        elements into <code>dstBuffer</code>, reading starting at <code>srcByteOffset</code>
-        and writing starting at <code>dstOffset</code> into <code>dstBuffer</code>.
+        Reads back data from the bound WebGLBuffer into <code>dstBuffer</code>.
+        <br><br>
+        Let <code>buf</code> be the buffer bound to <code>target</code>.
+        If <code>length</code> is 0, let <code>copyLength</code> be
+        <code>dstBuffer.length - dstOffset</code>; otherwise, let
+        <code>copyLength</code> be <code>length</code>.
+        <br><br>
+        Copy <code>copyLength</code> typed elements (each of size <code>dstBuffer.BYTES_PER_ELEMENT</code>)
+        from <code>buf</code> into <code>dstBuffer</code>,
+        reading <code>buf</code> starting at byte index <code>srcByteOffset</code> and
+        writing into <code>dstBuffer</code> starting at element index <code>dstOffset</code>.
+        If <code>copyLength</code> is 0, no data is written to <code>dstBuffer</code>.
```

Comment:
Does this fully specify all of the corner cases you ran into? Do we need to specify the case when copyLength is less than zero (though there are some signed/unsigned arithmetic issues in the expression above since dstOffset and length are both unsigned ints)?